gvr-picktest/app/src/androidTest/java/org/gearvrf/tester/GVRPickerTests.java

```diff
@@ -0,0 +1,159 @@
+            {
+                ++SphereHits;
+            }
+            else if (sceneObj.getName().equals("cube"))
+            {
+                ++CubeHits;
+            }
+        }
+        public void onExit(GVRSceneObject sceneObj)
+        {
+            sceneObj.getRenderData().getMaterial().setVec4("diffuse_color", 0, 0, 1, 1);
+        }
+        public void onNoPick(GVRPicker picker) { }
+        public void onPick(GVRPicker picker) { }
+        public void onInside(GVRSceneObject sceneObj, GVRPicker.GVRPickedObject pickInfo) { }
+    }
+
+    public GVRPickerTests() {
+        super();
+    }
+
+    @Rule
+    public ActivityTestRule<TestableActivity> mActivityRule = new
+            ActivityTestRule<TestableActivity>(TestableActivity.class);
+
+    @Before
+    public void setUp() throws TimeoutException
+    {
+        gvrTestUtils = new GVRTestUtils(mActivityRule.getActivity());
+        final Waiter waiter = new Waiter();
+        gvrTestUtils.setOnInitCallback(new OnInitCallback() {
+            public void onInit(GVRContext ctx)
+            {
+                GVRContext context = gvrTestUtils.getGvrContext();
+
+                mBlue = new GVRMaterial(context, GVRMaterial.GVRShaderType.BeingGenerated.ID);
+                mBlue.setDiffuseColor(0, 0, 1, 1);
+                mPickHandler = new PickHandler();
+                waiter.resume();
+            }
+        });
+        waiter.await(4000);
+    }
+
+    class PickCallback implements GVRTestUtils.OnRenderCallback
+    {
+        public GVRPicker Picker;
+        public GVRScene Scene;
+        public GVRSceneObject PickMe;
+
+        public PickCallback(GVRPicker picker, GVRScene scene, GVRSceneObject pickme)
+        {
+            Picker = picker;
+            Scene = scene;
+            PickMe = pickme;
+        }
+
+        public void onSceneRendered()
+        {
+            Picker.onDrawFrame(0);
+            GVRPicker.GVRPickedObject[] picked = Picker.getPicked();
+
+            Assert.assertNotNull(picked);
+            Assert.assertEquals(picked.length, 1);
+            Assert.assertEquals(picked[0].getHitObject(), PickMe);
+            Scene.removeSceneObject(PickMe);
+            Picker.setEnable(false);
+        }
+    }
+
+    @Test
+    public void canPickSphere()
+    {
+        GVRContext context = gvrTestUtils.getGvrContext();
+        GVRScene scene = gvrTestUtils.getMainScene();
+        GVRSceneObject sphere = new GVRSphereSceneObject(context, true, mBlue);
+        GVRSphereCollider collider = new GVRSphereCollider(context);
+
+        sphere.getRenderData().setShaderTemplate(GVRPhongShader.class);
+        sphere.getTransform().setPosition(0, 0, -2);
+        collider.setRadius(1.0f);
+        sphere.attachComponent(collider);
+        scene.addSceneObject(sphere);
+        GVRPicker picker = new GVRPicker(context, scene);
+        gvrTestUtils.setOnRenderCallback(new PickCallback(picker, scene, sphere));
```

Comment:
need to use waiter here for doing the asserts and then waiting. 
waiter.assertNotNull() in the pickerCallback
and 
wanter.resume(TIMEOUT) at 142 here.